### PR TITLE
feat(forms): Rename AppFormik to FormikEnhanced.

### LINKED ISSUE #77: [forms] Rename `useAppFormik` to `useFormikEnhanced`
It's confusing why it's called `useAppFormik` in the scope of the library. It's not returning `AppFormik`, it returns the same old `Formik`, it only accepts more arguments and has extra functionality under the hood.

### PR DESCRIPTION
**Pull request by doniyor2109**
Closes #77

### CODE REVIEW

File: packages/forms/src/FormikEnhancedProvider.tsx [hunk 1]
```diff
@@ -6,20 +6,20 @@ import React, {
   useMemo,
 } from 'react';
 
-export interface AppFormikContext {
+export interface FormikEnhancedContext {
```
Comment (umidbekkarimov) [inline packages/forms/src/FormikEnhancedProvider.tsx:9]: `useFormikEnhanced` is like `useFormik` but "enhanced". So I don't know where `FormikEnhancedContext` comes, maybe we should call it `FormsContext`?

File: packages/forms/src/useFormikEnhanced.ts [hunk 2]
```diff
@@ -25,18 +25,18 @@ export interface AppFormikConfig<TValues extends FormikValues, TResponse>
...
 }
 
-export type AppFormikStatus<TResponse> =
+export type FormikEnhancedStatus<TResponse> =
   | { type: 'initial'; payload?: undefined }
   | { type: 'submitted'; payload: TResponse }
   | { type: 'rejected'; payload: Error };
 
-export interface AppFormik<TValues extends FormikValues, TResponse>
+export interface FormikEnhanced<TValues extends FormikValues, TResponse>
```
Comment (umidbekkarimov) [inline packages/forms/src/useFormikEnhanced.ts:33]: We're not enhancing formik in any way, we just add typing for `status` and `setStatus`. It should `FormikContextTypeWithTypedStatus`, and with some todo that says that this should be removed once https://github.com/jaredpalmer/formik/pull/2323 will be merged and released.

Comment (doniyor2109) [reply packages/forms/src/useFormikEnhanced.ts:33]: Maybe `FormikContextTypeEnhanced` ?

Comment (umidbekkarimov) [reply packages/forms/src/useFormikEnhanced.ts]: @doniyor2109 I like it!

**Pull request merged**